### PR TITLE
Assign and grade group branch

### PR DESCRIPTION
**Pull request by lsyurea**
## Description

This pull request aims to implement an assignment feature that allows for the efficient assignment of the same task to a group of persons with the maximum available marks.

## Context

This feature is valuable for optimizing the assignment process, especially in scenarios where multiple persons in a group should receive the same assignment with the highest possible marks. This addition simplifies the workflow for our users and improves the overall efficiency of our system.

## Checklist

Please make sure to review and complete the following checklist items:

- [X] I have thoroughly self-reviewed my changes to ensure accuracy and functionality.
- [X] I have provided comprehensive JavaDocs for all relevant methods to enhance code documentation.
- [X] I have updated the documentation, including:
  - [X] User Guide: Modified the Table of Contents, Description, and Summary sections to reflect these changes.
  - [X] Developer Guide: Updated if applicable to include details about this new feature.
- [X] I have executed `./gradlew test` or `./gradlew check`, and all checks have successfully passed without errors.
- [X] I have updated my project portfolio with clear details of my contributions related to this pull request.

Feel free to provide your feedback and insights during the review process. Thank you.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AssignmentGroupCommand.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ */
+public class AssignmentGroupCommand extends Command {
+
+    public static final String COMMAND_WORD = "assignGroup";
+    public static final String MESSAGE_SUCCESS = "Assignment created successfully!";
+    public static final String MESSAGE_FAIL = "Assignment not created.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Creates an assignment for students in the same group.\n"
+            + "Compulsory Parameters: "
+            + PREFIX_NAME + "ASSIGNMENT_NAME " + "GROUP " + PREFIX_MAX_SCORE + "MAX_SCORE "
```
Comment (et-irl) [inline src/main/java/seedu/address/logic/commands/AssignmentGroupCommand.java:30]: The parameters seem to be in the wrong order.

File: src/main/java/seedu/address/logic/commands/AssignmentGroupCommand.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Executes the command to create a new assignment for all students in the group.
+     * @param model Model representing the current state of the address book.
+     * @return CommandResult representing the result of the execution.
+     * @throws CommandException If there is an error during command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList().filtered(person -> person.getGroup().equals(group));
```
Comment (et-irl) [inline src/main/java/seedu/address/logic/commands/AssignmentGroupCommand.java:65]: If the tutorial group does not exist, the command will still go through, giving assignments to nobody. Is this expected behaviour?

**Review by et-irl — COMMENTED** [review]
Looks good other than a few nitpicks!

File: docs/UserGuide.md
Comment (et-irl) [inline docs/UserGuide.md:1]: Your commands seem to be missing from the table of contents.

**Review by et-irl — COMMENTED** [review]
One more nit. Otherwise, good job!

File: docs/UserGuide.md
Comment (lsyurea) [reply docs/UserGuide.md:1]: Thanks, I have resolved it.

**Review by et-irl — APPROVED** [review]
Looking great to myself! (I am looking great.)

**Pull request merged**